### PR TITLE
[#IOPLT-463] Add Base resources for subscription flow

### PR DESCRIPTION
**Pull request by AleDore**
#### Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] I have committed only changes relevant to the task.
- [ ] I have minimized the number of changes.
- [ ] My changes are about only one task or issue.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Krusty93** [comment]
I strongly suggest to stick to the [naming convention](https://pagopa.atlassian.net/wiki/spaces/DevEx/pages/1032880258/Azure+Naming+convention+risorse+Italy+North?focusedCommentId=1038680400) we are adopting for new resources, in order to make them more clearer, coincise and maintainable over time. Also, following [this tutorial](https://pagopa.atlassian.net/wiki/spaces/DevEx/pages/1035272271/Configurazione+di+un+repository+GitHub) would help this repository to grow adopting existing reusable tools, following a standards already in use by other repos.

Regarding TF, do you see any advantage adopting one of the [recommended approaches](https://pagopa.atlassian.net/wiki/spaces/ENG/pages/910492147/RFC-ENG+Riorganizzazione+configurazioni+Terraform) to organize TF files?

File: infra/src/database.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  private_endpoint_sql_name           = "${local.project}-citizen-auth-account"
+  private_service_connection_sql_name = "${local.project}-citizen-auth-account-private-endpoint"
+  private_dns_zone_sql_ids            = [data.azurerm_private_dns_zone.privatelink_documents_azure_com.id]
+  subnet_id                           = data.azurerm_subnet.private_endpoints_subnet.id
+  is_virtual_network_filter_enabled   = false
+
+  main_geo_location_location       = azurerm_resource_group.data_rg.location
+  main_geo_location_zone_redundant = true
+  additional_geo_locations = [{
+    location          = "northeurope"
```
Comment (Krusty93) [inline infra/src/database.tf:29]: Shouldn't be `gwc` the `itn` secondary region?

File: infra/src/network.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  name                 = "${local.project}-pendpoints-snet"
+  address_prefixes     = var.snet_pendpoints_address_spaces
+  resource_group_name  = azurerm_resource_group.net_rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+
+  private_endpoint_network_policies_enabled     = false
+  
+}
+
+resource "azurerm_private_dns_zone" "privatelink_documents" {
```
Comment (Krusty93) [inline infra/src/network.tf:31]: Probably we can use the existing private dns zone and link it to this new vnet?

File: infra/src/99_variables.tf [hunk 3]
```diff
@@ -31,4 +31,72 @@ variable "tags" {
...
+
+variable "cidr_subnet_fnsubscriptionasync" {
+  type    = list(string)
+  description = "The subscription async function address space"
+}
+
+variable "cidr_subnet_fnsubscription" {
+  type    = list(string)
+  description = "The subscription function address space"
+}
```
Comment (AleDore) [inline infra/src/99_variables.tf:53]: please @Krusty93 can you check addresses in order to avoid overlap with other VNET address spaces?

**Pull request merged**